### PR TITLE
[WIP] allow to add URLs as remote source for task where they redirect to the actual resource location

### PR DESCRIPTION
**Pull request by japrescott**
### Motivation and context
When adding a remote source URL for a task and that URL makes a redirect (e.g first auths the user and then forwards it to where the resource is located), the file wont be downloaded and not imported. By allowing to follow redirects when trying to download the resource, this limitation is removed

### How has this been tested?
Ran it locally and tested it

### Checklist

- [X] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [X] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [X] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat/apps/engine/task.py [hunk 1]
```diff
@@ -257,7 +257,7 @@ def _download_data(urls, upload_dir):
         job.meta['status'] = '{} is being downloaded..'.format(url)
         job.save_meta()
 
-        response = requests.get(url, stream=True)
+        response = requests.get(url, stream=True, allow_redirects=True)
```
Comment (nmanovic) [inline cvat/apps/engine/task.py:260]: @azhavoro , do you think it introduces any security risks? Will the code above validate redirected URL?

Comment (nmanovic) [inline cvat/apps/engine/task.py:260]: @azhavoro , do you think it introduces any security risks? Will the code above validate redirected URL?

Comment (azhavoro) [reply cvat/apps/engine/task.py:260]: There is SSRF volnureability, in case of original URL response retruns redirect to localhost for example. The function above is not follow to HTTP redirects and validate only original URL based on domain/IP address.

I think there are 2 option:
1. Handle all redirects manually
2. Don't change this and allow only "direct" links

@japrescott What do you think?

Comment (japrescott) [reply cvat/apps/engine/task.py:260]: @azhavoro Good catch with SSRF. But I think the vulnerability is small;
- if a user of CVAT wants to mess with CVAT internals, it can just do so directly by entering `http://localhost:8080/somewhere` already (no need for redirecting URLS)
- if a user trusts a bad website and adds a URL which would try to do something malicious, maximally a bad image could be injected into CVAT

Manually handling redirects does not mitigate SSRF and direct links are way to limiting (think Auth, JWT, etc)

In conclusion; I would allow these changes with `allow_redirects=True`.

Comment (azhavoro) [reply cvat/apps/engine/task.py:260]: @japrescott Hi, sorry for the late response

> @azhavoro Good catch with SSRF. But I think the vulnerability is small;

I don't think so if we are talking about publicly accessed intansers

> * if a user of CVAT wants to mess with CVAT internals, it can just do so directly by entering `http://localhost:8080/somewhere` already (no need for redirecting URLS)

Can you make a get request to all internal cvat services like elasticsearch this way? But if follow_redirect is enabled it's pretty easy

> * if a user trusts a bad website and adds a URL which would try to do something malicious, maximally a bad image could be injected into CVAT

Agree, but this case is under of scope of SSRF vulnerability.

So, I think `allow_redirects=True` by default it is not a good idea.

**Comment by nmanovic** [comment]
@japrescott , due to security risks, I will recommend to implement the manual handling of redirects and validate every redirected URL.

**Comment by nmanovic** [comment]
@japrescott , I will close the PR for now. Don't hesitate to re-open it if you want to fix it in more secure way.
